drivers/can/can_mcp2515.c

```diff
@@ -736,10 +736,12 @@ static void mcp2515_handle_errors(const struct device *dev)
 }
 
 #ifndef CONFIG_CAN_AUTO_BUS_OFF_RECOVERY
-static void mcp2515_recover(const struct device *dev, k_timeout_t timeout)
+static int mcp2515_recover(const struct device *dev, k_timeout_t timeout)
 {
 	ARG_UNUSED(dev);
 	ARG_UNUSED(timeout);
+
+	return -ENOTSUP;
 }
```

Comment:
note: shouldn't this function implementation be optional? if NULL, API could then return `-ENOSYS`

Reply:
I would rather not make it optional. It is common functionality that _should_ be supported by all drivers/devices.

The MCP2515 is a bit special (also in this regard) as it does not offer a register setting for selecting between manual and automatic bus recovery. The datasheet says that this _can_ be implemented by hooking into the error ISR, but this is not implemented in the Zephyr MCP2515 driver (yet?).